src/Build/Definition/ProjectCollection.cs

```diff
@@ -1419,10 +1336,7 @@ public void UnloadAllProjects()
         /// </summary>
         public ProjectPropertyInstance GetGlobalProperty(string name)
         {
-            using (_locker.EnterDisposableUpgradeableReadLock())
-            {
-                return _globalProperties[name];
-            }
+            return _globalProperties[name];
```

Comment:
`_globalProperties` is thread-safe so it doesn't look like a single access like this has to run under `_lockerGlobalProperties`. It would be less fragile to do it, though. Will fix.